### ISSUE TITLE
[CurlHttpClient] Problem with the ssl ca cert (path? access rights?)

### ISSUE THREAD

**Issue by Bryan988** [symfony/symfony-docs#15440]
Programs | Version
------------ | -------------
Symfony | 4.4
PHP | 7.4.9
Windows | 10

## Description

I am using [Multiplex Responses](https://symfony.com/doc/4.4/http_client.html#multiplexing-responses) as described in the documentation. This is working fine for small amount of request like 400 / 500. 
Unfortunately, when I start to do more that 500 requests, I get the following error displayed : 
```
Error : problem with the ssl ca cert (path? access rights?)
``` 
In the curl debug, this is displayed : 
```
error setting certificate verify locations:
CAfile: "C:\cacert\cacert.pem
CApath : none
```

The CAfile is working fine, The issue might come from the maximum number of simultaneous processes reading the same file.
The requests are all different.

## Possible Solution 

Loop on a smaller number of requests reset the client before beginning the loop.
After looking at `CurlHttpClient.php` . I spotted a method called reset(). 
The problem is that I have already tried this solution and it raises another issue. 
```
In CurlResponse.php line 275:
 [ErrorException]                                      
  Warning: Invalid callback , no array or string given 
```

Curlresponse.php : 
```
272  try {
273            self::$performing = true;
274            $active = 0;
275            while (\CURLM_CALL_MULTI_PERFORM === curl_multi_exec($multi->handle, $active));

```

## Working

By limiting the max number of requests to 500, and without the `reset()` method, it is working. My issue is that it is too limited as I have to do more that 100k requests. It will be too long

**Comment by javiereguiluz** [comment]
@Bryan988 thanks for the detailed explanation of the issue. It will be helpful!

Let's ask to @nicolas-grekas  do you think this might be caused by some issue of the computer running this code (e.g. max simultaneous reads of the same file)? Or maybe it's a constraint of cURL or Symfony HttpClient? Or maybe a "bug" or something we can improve in the code? Thanks!

**Comment by nicolas-grekas** [comment]
I'm not aware of this issue. SO has this page: https://stackoverflow.com/questions/27987091/ssl-ca-cert-path-access-rights
Google might find more possible reasons.

Anyway, it's a good idea to process requests in batches. I started in https://github.com/symfony/symfony/pull/38690 but stopped and closed after (I explained why in the PR).

TLDR; do the batching on your side :)
